core/templates/pages/exploration-player-page/templates/lesson-information-card-modal.component.ts

```diff
@@ -0,0 +1,150 @@
+  // where we need to do non-null assertion. For more information see
+  // https://github.com/oppia/oppia/wiki/Guide-on-defining-types#ts-7-1
+  storyTitleTranslationKey!: string;
+  storyPlaythroughObject!: StoryPlaythrough;
+  storyId!: string;
+  storyTitle: string = '';
+  topicUrlFragment: string;
+  classroomUrlFragment: string;
+  storyUrlFragment: string;
+  expTitleTranslationKey!: string;
+  expDescTranslationKey!: string;
+  displayedCard!: StateCard;
+  explorationId!: string;
+  expTitle!: string;
+  expDesc!: string;
+  contributorNames!: string[];
+  hasStoryTitle!: boolean;
+  chapterTitle!: string;
+  chapterDesc!: string;
+  chapterNumber!: string;
+  numberofCheckpoints!: number;
+  expInfo: LearnerExplorationSummaryBackendDict;
+  startedWidth: number = 0;
+  completedWidth!: number;
+  separatorWidth: number = 0;
+  separatorArray: number[] = [];
+
+
+  constructor(
+    private ngbModal: NgbModal,
+    private ngbActiveModal: NgbActiveModal,
+    private urlService: UrlService,
+    private i18nLanguageCodeService: I18nLanguageCodeService,
+    private storyViewerBackendApiService: StoryViewerBackendApiService,
+  ) {
+    super(ngbActiveModal);
+  }
+
+  ngOnInit(): void {
+    this.explorationId = this.expInfo.id;
+    this.expTitle = this.expInfo.title;
+    this.expDesc = this.expInfo.objective;
+    this.storyId = this.urlService.getUrlParams().story_id;
+    this.hasStoryTitle = (this.storyId !== undefined) ? true : false;
+
+    this.storyTitleTranslationKey = (
+      this.i18nLanguageCodeService
+        .getStoryTranslationKey(
+          this.storyId, TranslationKeyType.TITLE));
+
+    this.expTitleTranslationKey = (
+      this.i18nLanguageCodeService.
+        getExplorationTranslationKey(
+          this.explorationId, TranslationKeyType.TITLE)
+    );
+    this.expDescTranslationKey = (
+      this.i18nLanguageCodeService.
+        getExplorationTranslationKey(
+          this.explorationId, TranslationKeyType.DESCRIPTION)
+    );
+
+    if (this.hasStoryTitle) {
+      this.topicUrlFragment = (
+        this.urlService.getTopicUrlFragmentFromLearnerUrl());
+      this.classroomUrlFragment = (
+        this.urlService.getClassroomUrlFragmentFromLearnerUrl());
+      this.storyUrlFragment = (
+        this.urlService.getStoryUrlFragmentFromLearnerUrl());
+
+      this.storyViewerBackendApiService.fetchStoryDataAsync(
+        this.topicUrlFragment,
+        this.classroomUrlFragment,
+        this.storyUrlFragment).then(
+        (storyDataDict) => {
+          this.storyTitle = storyDataDict.title;
+        });
+    }
+
+
+    // The purpose of separatorArray is for the working of
+    // for loop in lesson-information-card-modal.component.html.
+    this.separatorArray = new Array(this.numberofCheckpoints);
+    this.separatorWidth = 100 / (this.numberofCheckpoints);
+
+    this.startedWidth = 100 / (this.numberofCheckpoints);
+    if (this.completedWidth === 100) {
+      this.startedWidth = 0;
+    }
```

Comment:
Since these are derived from numberOfCheckpoints, should we consider doing the calculation inline in the template instead?

Reply:
It's possible, but wouldn't that then escape from tests and also complicate the template file?

Reply:
From what I can tell, these seem to be related to the presentation aspect of the progress bar (e.g. width, separation), so it seems like something that can go in to the template. The component can focus on logic / functionality.

I think we will need to have e2e tests for this.